metricbeat/module/mongodb/replstatus/info.go

```diff
@@ -43,85 +43,109 @@ type CollSize struct {
 
 const oplogCol = "oplog.rs"
 
+// getReplicationInfo returns oplog info from local.oplog.rs
 func getReplicationInfo(client *mongo.Client) (*oplogInfo, error) {
-	// get oplog.rs collection
+	// Get oplog collection info from local.oplog.rs (<db>.<collection>)
 	db := client.Database("local")
 
-	collections, err := db.ListCollectionNames(context.Background(), bson.D{})
-	if err != nil {
-		return nil, fmt.Errorf("could not retrieve collection names: %w", err)
-	}
-
-	if !contains(collections, oplogCol) {
-		return nil, errors.New("collection oplog.rs was not found")
+	// NOTE(shmsr):
+	// https://www.mongodb.com/docs/manual/reference/command/collStats/#syntax
+	// "scale" field is omitted here as it is by default 1, i.e., it return sizes in bytes.
+	//
+	// Also, note that collStats is deprecated since v6.2 but as we support older
+	// versions i.e., >= 5.0, let's keep it for now as this still works.
+	// TODO(shmsr): For newers versions, we can use db.collection.stats()
+	// https://www.mongodb.com/docs/manual/reference/method/db.collection.stats/#mongodb-method-db.collection.stats
+	// or use this: https://github.com/percona/mongodb_exporter/blob/95d1865e34940d0d610bb1fbff9745bc66ddbc73/exporter/collstats_collector.go#L100
+	res := db.RunCommand(context.Background(), bson.D{
+		{Key: "collStats", Value: oplogCol},
```

Comment:
This would work for v6.2 onwards as well ?

Reply:
Or do we need a version based implementation of db.collection.stats() here as mentioned in your comment ?

Reply:
No it works. Mongosh (mongosh uses the very same thing for oplog window).